### PR TITLE
Fix IN(...) clause doesnt permit multiple values

### LINKED ISSUE #16041: Search IN(...) Clause doest not permit multiple values
### Describe the bug

Not sure but it seems to be the same issue as in #10105. (Which has been resolved in 2015)

### To Reproduce

1. Go to my database and select a table
2. Click on Search
3. In my Column "ID" I select the Operator "IN(...)"
3. In my Column "ID" I enter the value "1,2,3"
4. And the error shows "Please enter a valid number."

### Expected behavior

No error, hit search and get the result.

### Screenshots

Screenshot 2020-03-25 at 10 53 41

### Server configuration

- Operating system: Ubuntu 18.04
- Web server: Apache2
- Database version: 5.7.29
- PHP version: 7.4.4
- phpMyAdmin version: 5.0.2

### Client configuration

- Browser: Google Chrome
- Operating system: MacOS

### Additional context

n/a

### PR DESCRIPTION
**Pull request by ankush-0x90**
### Description

Please describe your pull request.
Fixes #16041 
Added regex validation for Multiple Clauses Like
IN(...) NOT IN(...) BETWEEN 

Before submitting pull request, please review the following checklist:

- [x] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [x] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [x] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [x] Every commit has a descriptive commit message.
- [x] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [ ] Any new functionality is covered by tests.

### CODE REVIEW

File: js/table/change.js [hunk 1]
```diff
@@ -153,13 +153,46 @@ function checkForCheckbox (multiEdit) {
...
-        $('#tbl_search_form').validate();
-        validateIntField($thisInput, true);
+        if (hasMultiple) {
+            $('#tbl_search_form').validate();
+            // validator method for IN(...), NOT IN(...)
+            // BETWEEN and NOT BETWEEN
+            jQuery.validator.addMethod('validationFunctionForMultipleInt', function (value) {
+                return value.match(/^(\d\s*)+(,\s*\d+)*$/i) !== null;
+            },
+            'Please enter a valid numbers.'
```
Comment (williamdes) [inline js/table/change.js:166]: You should use https://github.com/phpmyadmin/phpmyadmin/blob/09c1d1070a8027c6572a6eb77e39f91bdd9b2b75/js/messages.php#L123

Comment (asprazz) [reply js/table/change.js:166]: Done..

File: js/table/change.js [hunk 2]
```diff
@@ -153,13 +153,46 @@ function checkForCheckbox (multiEdit) {
 // eslint-disable-next-line no-unused-vars
 function verifyAfterSearchFieldChange (index) {
     var $thisInput = $('input[name=\'criteriaValues[' + index + ']\']');
+    var hasMultiple = $thisInput.prop('multiple');
     // validation for integer type
     if ($thisInput.data('type') === 'INT') {
```
Comment (williamdes) [inline js/table/change.js:158]: Consider moving hasMultiple into this if condition to narrow the use cases of this variable :)

File: js/table/change.js [hunk 3]
```diff
@@ -756,8 +789,25 @@ function changeValueFieldType (elem, searchIndex) {
         'BETWEEN' === type ||
         'NOT BETWEEN' === type
     ) {
-        $('#fieldID_' + searchIndex).prop('multiple', true);
+        var mini = $('#fieldID_' + searchIndex).attr('min');
```
Comment (williamdes) [inline js/table/change.js:792]: Could all this operations be done in the PHP code ?

**Review by williamdes — CHANGES_REQUESTED** [review]
95%okay for me
Good work
I need to test it after your changes

File: js/table/change.js (hunk 2 above)
Comment (asprazz) [reply js/table/change.js:158]: Oke Got It!

File: js/table/change.js (hunk 3 above)
Comment (asprazz) [reply js/table/change.js:792]: I will try ...

Comment (williamdes) [reply js/table/change.js:792]: :muscle:

Comment (asprazz) [reply js/table/change.js]: I have changed the attributes to data-min and data-max in the backend so IN(...) , NOT IN(....) clauses won't have that problem of min max I hope it works for us! :)

File: js/table/change.js [hunk 4]
```diff
@@ -756,8 +789,8 @@ function changeValueFieldType (elem, searchIndex) {
         'BETWEEN' === type ||
         'NOT BETWEEN' === type
     ) {
-        $('#fieldID_' + searchIndex).prop('multiple', true);
+        $('#fieldID_' + searchIndex).attr('multiple', true);
     } else {
-        $('#fieldID_' + searchIndex).prop('multiple', false);
```
Comment (asprazz) [inline js/table/change.js:761]: Should i revert this to `prop` also ?

Comment (asprazz) [reply js/table/change.js]: I reverted... it back to `prop`

**Review by williamdes — APPROVED** [review]
:100:

File: js/table/change.js (hunk 3 above)
Comment (williamdes) [reply js/table/change.js]: Very nice, thank you

File: libraries/classes/Controllers/Table/SearchController.php [hunk 5]
```diff
@@ -974,8 +974,8 @@ public function getColumnProperties($search_index, $column_index)
                 $cleanType,
                 ! $is_unsigned
             );
-            $html_attributes = 'min="' . $min_max_values[0] . '" '
-                            . 'max="' . $min_max_values[1] . '"';
+            $html_attributes = 'data-min="' . $min_max_values[0] . '" '
```
Comment (williamdes) [inline libraries/classes/Controllers/Table/SearchController.php:977]: FTR: Line added by e92e51d42d38b9516153c011781bc90f94c144d5

Comment (asprazz) [reply libraries/classes/Controllers/Table/SearchController.php:977]: Yeas!

**Comment by williamdes** [comment]
@asprazz I tested your work and a minor issue was introduced.
When you will the field and the validation does pass, then you empty the input and it stays invalid as it is empty.
As a user I would like to ba able to empty the input and the validation resets.

Could you please fix this behavior ?

**Comment by asprazz** [comment]
Hi @williamdes I will look into it asap.

**Review by williamdes — APPROVED** [review]

**Comment by williamdes** [comment]
Thank you @asprazz for fixing this issue !

**Comment by asprazz** [comment]
> Thank you @asprazz for fixing this issue !

I’m glad I could help. :man_technologist:

**Pull request merged**